packages/bot-handler-avs/src/AVSHandler.ts

```diff
@@ -0,0 +1,186 @@
+    return wCall;
+  }
+
+  private readonly mediaHandler = async (): Promise<MediaStream> => {
+    return this.mediaStream;
+  };
+
+  private readonly createWUser = (wCall: Wcall, selfUserId: string, selfClientId: string): number => {
+    /* cspell:disable */
+    const wUser = wCall.create(
+      selfUserId,
+      selfClientId,
+      () => {},
+      this.onSendCallMessage,
+      this.sendSFTRequest,
+      this.onIncomingCall,
+      () => {},
+      () => {},
+      () => {},
+      this.callClosed,
+      this.callConfigRequestHandler,
+      () => 0,
+      () => {},
+      () => {},
+    );
+
+    wCall.setParticipantChangedHandler(wUser, () => {});
+    return wUser;
+  };
+
+  private callConfigRequestHandler(convid: string, metrics_json: string, arg: number) {
+    this.account!.service!.account.getCallConfig()
+      .then((config: CallConfigData) => {
+        this.wCall!.configUpdate(this.wUser!, 0, JSON.stringify(config));
+      })
+      .catch(console.error);
+
+    return 0;
+  }
+
+  private readonly sendSFTRequest = (
+    context: number,
+    url: string,
+    data: string,
+    dataLength: number,
+    _: number,
+  ): number => {
+    void (async () => {
+      try {
+        const response = await axios.post(url, data);
+
+        const {status, data: axiosData} = response;
+        const jsonData = JSON.stringify(axiosData);
+        this.wCall!.sftResp(this.wUser!, status, jsonData, jsonData.length, context);
+      } catch (error) {
+        console.info('Failed to send SFT request', error);
+      }
+    })();
+    return 0;
+  };
+
+  private readonly onSendCallMessage = (
+    ctx: number,
+    conversationId: string,
+    selfUserId: string,
+    selfClientId: string,
+    userid_dest: string | undefined,
+    clientid_dest: string | undefined,
+    data: string,
+  ): number => {
+    void (async () => {
+      const callPayload = this.account!.service!.conversation.messageBuilder.createCall(conversationId, data);
+      await this.account!.service!.conversation.send(callPayload);
+    })();
+
+    return 0;
+  };
+
+  private readonly onIncomingCall = (
+    conversationId: string,
+    timestamp: number,
+    userId: string,
+    clientId: string,
+    hasVideo: number,
+    shouldRing: number,
+  ) => {
+    const call: Call = {
+      clientId,
+      conversationId,
+      hasVideo,
+      shouldRing,
+      timestamp,
+      userId,
+    };
+    this.storeCall(call);
+  };
+
+  private readonly callClosed = (reason: REASON, conversationId: string) => {
+    this.removeCall(conversationId);
+  };
+
+  private readonly storeCall = (call: Call) => {
+    this.activeCalls[call.conversationId] = call;
+  };
+
+  private readonly removeCall = (conversationId: string) => {
+    delete this.activeCalls[conversationId];
+  };
+
+  async startCall(conversationId: string, conversationType: CONV_TYPE, callType: CALL_TYPE): Promise<void> {
+    if (this.wCall && typeof this.wUser === 'number') {
```

Comment:
Would be nice to have this inverted `if wCall or wUser doesn't match expectations -> throw error`